snmp/datadog_checks/snmp/snmp.py

```diff
@@ -280,14 +277,14 @@ def _profile_for_sysobject_oid(self, sys_object_oid):
         """
         Return the most specific profile that matches the given sysObjectID.
         """
-        profiles = [profile for oid, profile in self.profiles_by_oid.items() if fnmatch.fnmatch(sys_object_oid, oid)]
+        matched_oids = [
+            (oid, profile) for oid, profile in self.profiles_by_oid.items() if fnmatch.fnmatch(sys_object_oid, oid)
+        ]
 
-        if not profiles:
+        if not matched_oids:
             raise ConfigurationError('No profile matching sysObjectID {}'.format(sys_object_oid))
 
-        return max(
-            profiles, key=lambda profile: oid_pattern_specificity(self.profiles[profile]['definition']['sysobjectid'])
-        )
+        return max(matched_oids, key=lambda oid: oid[0])[1]
```

Comment:
Don't we have to keep `oid_pattern_specificity()` here?

Genuinely asking the question. Originally this was introduced because we found a weird edge case in which lexicographic order was not enough. Looking at the unit tests for `oid_pattern_specificity` it looks to be when profiles from different OID sub-trees match, but that's not possible because a device can only match profiles whose OID are in a common sub-tree. So hmm…

Anyway IMO this should be a separate PR referencing https://github.com/DataDog/integrations-core/pull/5813 (which added the `oid_pattern_specificity` thing).

Reply:
That's strange indeed.. I will keep `oid_pattern_specificity` for this PR